### DISCUSSION TITLE
Streaming JSON objects with one JSON object per HTTP chunk - Streaming for lines oriented formats (NDJSON, CSV, etc)

### DISCUSSION THREAD

**Discussion by kael** [remix-run/remix#5824] [Q&A]
With SSR streaming, I'm trying to (transform and) stream JSON objects so that each JSON object is sent in its own HTTP streamed chunk.

Objects are returned in an array from an API. I'd like to use [`JSON Stream`](https://workers.tools/json-stream) to slice objects from the array similarly to this example:

```js
const collected = [];
await (await fetch('/nested.json')).body
  .pipeThrough(new JSONParseStream('$.items.*'))
  .pipeTo(new WritableStream({ write(obj) { collected.push(obj) }}))
```

I'm looking to emulate something visually similar to [Suspense streaming](https://suspense.vercel.app/guide/writing-a-streaming-cache).

Can the data loader or the fetcher stream the response from the loader by handling the HTTP `Response#body` property ?

I tried several combinations, also with `defer`, but the best I can get from the data loader is a HTTP `Response#body` property but an unusable one (an empty object) when I pass a `TransformStream.readable` object to the returned `Response` in the loader.

Can Remix handle streams between the loader and the function component ? How can Remix handle [NDJSON](https://andreitopli.medium.com/working-with-newline-delimited-json-using-can-ndjson-stream-in-typescript-983016d51d4a) or CSV so that each data entity (object, line, etc.) is sent in its own HTTP chunk instead of having the full array of objects sent (as seen with Wireshark when inspecting the TCP stream) ?

**Comment by DarkPurple141** [comment]
I’m also keen for this that has a unconsumed body from a second server which it would be cool to pass off to directly a stream rather than consuming and then re-sending.
